### DISCUSSION TITLE
How to set the response headers in a Vike standalone server (Not vike as a middleware)

### DISCUSSION THREAD

**Discussion by Blankeos** [vikejs/vike#1905] [Discussion] [answered]
I noticed there's no way to set the response headers using the `pageContext` object in `+data.ts`.

The only way I know currently is to use Vike as a middleware and by passing it in `pageContextInit` then `renderPage(pageContextInit)`

Also noticed `pageContext.httpResponse` doesn't exist in `+data.ts` either.

How do you guys do it in a standalone Vike app?

**Comment by brillout** [comment, marked as answer]
See https://github.com/vikejs/vike/issues/1803.

**Reply by Blankeos** [reply]
Thanks Rom! Watching this.
